stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -141,6 +141,9 @@ Random.seed!(1)
                 @test rdiv!(Uc, D) ≈ target atol=atol_three
                 @test_throws DimensionMismatch rdiv!(Matrix{elty}(I, n-1, n-1), D)
                 @test_throws SingularException rdiv!(Uc, Diagonal(fill!(similar(D.diag), 0)))
+                @test rdiv!(Diagonal([2.0,3.0]), Diagonal(2:3)) == Diagonal([1.0,1.0])
+                @test rdiv!(Diagonal([2,3]), Diagonal(2.0:3.0)) == Diagonal([1.0,1.0])
+                @test rdiv!(Diagonal([2,3]), Diagonal(2:3)) == Diagonal([1,1])
```

Comment:
Should the tests all be Diagonal Diagonal? 

Also, they are inside a loop `for (v, U) in ((vv, UU), (view(vv, 1:n), view(UU, 1:n, 1:2)))` but don't use that. Which is itself inside a loop `for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})`. I'm not too sure what all these loops are doing, but perhaps worth understanding right way to test the objects/types the loop is concerned with, or why not to be concerned.